File: backends/backend-springboot/src/main/java/com/talentsphere/backend/model/Lesson.java
package com.talentsphere.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "lessons")
@Data
@NoArgsConstructor
@EntityListeners(AuditingEntityListener.class)
public class Lesson {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "section_id", nullable = false)
    private Section section;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private LessonType type;

    @Column(nullable = false)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "order_index", nullable = false)
    private Integer orderIndex;

    @Column(name = "video_url")
    private String videoUrl;

    private Integer duration;

    @Column(name = "content_markdown", columnDefinition = "TEXT")
    private String contentMarkdown;

    @Column(name = "challenge_id")
    private UUID challengeId; // Loose coupling to Challenge entity

    @Column(name = "is_active")
    private boolean isActive = true;

    @CreatedDate
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    public enum LessonType {
        video, quiz, challenge, text
    }
}

File: backends/backend-dotnet/Models/Lesson.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TalentSphere.API.Models
{
    public class Lesson
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid Id { get; set; }

        public Guid SectionId { get; set; }
        [ForeignKey("SectionId")]
        public Section? Section { get; set; }

        [Required]
        public string Type { get; set; } = "video"; // video, quiz, challenge, text

        [Required]
        public string Title { get; set; } = string.Empty;

        public string? Description { get; set; }
        public int OrderIndex { get; set; }

        public string? VideoUrl { get; set; }
        public int? Duration { get; set; }
        public string? ContentMarkdown { get; set; }
        public Guid? ChallengeId { get; set; }

        public bool IsActive { get; set; } = true;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }
}
